docker/lookaside.go

```diff
@@ -175,6 +182,13 @@ func (config *registryConfiguration) signatureTopLevel(ref dockerReference, writ
 	if config.DefaultDocker != nil {
 		logrus.Debugf(` Using "default-docker" configuration`)
 		if url := config.DefaultDocker.signatureTopLevel(write); url != "" {
+			if uid := uid.GetRootlessUID(); uid != 0 {
+				if url != defaultDockerDir {
+					// use user configured default-docker
+					return url
+				}
+				return filepath.Join(homedir.Get(), defaultRootlessDockerDir)
```

Comment:
It would be rather surprising if the value visible in the config file (and only one specific value) were silently overridden. The conversation in #1027 suggested that:
- We would ship config files without a default (or no config file at all); but users upgrading from an older release could/would retain the old configuration
- The code would contain a built-in default that only applied if the config files did not specify anything.

Also, the man page should be updated.

Reply:
@mtrmac  PTAL.  Use the old configuration if the default configuration exists. And when euid not 0, only use the built-in default if there's no per-user configuration.